statsmodels/tsa2/sarimax/estimators/tests/test_gls.py

```diff
@@ -0,0 +1,237 @@
+
+from statsmodels.tsa2.datasets.brockwell_davis_2002 import lake, oshorts
+from statsmodels.tsa2.sarimax.estimators.gls import gls
+
+@pytest.mark.low_precision('Test against Example 6.6.1 in Brockwell and Davis'
+                           ' (2016)')
+def test_brockwell_davis_example_661():
+    # Get the overshorts data
+    endog = oshorts.copy()
+
+    exog = np.ones_like(endog)
+
+    # Here we restrict the iterations to 1 and test against the values in the
+    # text (set tolerance=1 to suppress to warning that it didn't converge)
+    res, _ = gls(endog, exog, order=(0, 0, 1), max_iter=1, tolerance=1)
+    assert_allclose(res.exog_params, -4.745, atol=1e-3)
+    assert_allclose(res.ma_params, -0.818, atol=1e-3)
+    assert_allclose(res.sigma2, 2041, atol=1)
+
+    # Here we do not restrict the iterations and test against the values in
+    # the last row of Table 6.2 (note: this table does not report sigma2)
+    res, _ = gls(endog, exog, order=(0, 0, 1))
+    assert_allclose(res.exog_params, -4.780, atol=1e-3)
+    assert_allclose(res.ma_params, -0.848, atol=1e-3)
+
+
+@pytest.mark.low_precision('Test against Example 6.6.2 in Brockwell and Davis'
+                           ' (2016)')
+def test_brockwell_davis_example_662():
+    # Get the lake data
+    endog = lake.copy()
+
+    exog = np.c_[np.ones_like(endog), np.arange(1, len(endog) + 1)]
+
+    res, _ = gls(endog, exog, order=(2, 0, 0))
+
+    # Parameter values taken from Table 6.3 row 2, except for sigma2 and the
+    # last digit of the exog_params[0], which were given in the text
+    assert_allclose(res.exog_params, [10.091, -.0216], atol=1e-3)
+    assert_allclose(res.ar_params, [1.005, -.291], atol=1e-3)
+    assert_allclose(res.sigma2, .4571, atol=1e-3)
+
+
+@pytest.mark.low_precision('Test against Example 6.6.3 in Brockwell and Davis'
+                           ' (2016)')
+# @pytest.mark.xfail(reason="Source appears to find suboptimal parameters")
+def test_brockwell_davis_example_663():
+    # TODO: the parameters described by BD appear to be suboptimal (based on
+    # llf computed from state space form), so that this test fails. Should try
+    # to confirm with ITSM2000 (i.e. see if we can get it to find better
+    # parameters closer to what we find, or compare llf, or something).
+    # TODO: quite a slow test, and xfail anyway due to finding better
+    # parameters, so commented out
+    pass
+
+    # Get the data, perform seasonal differencing
+    # endog = sbl.diff(12).iloc[12:]
+
+    # exog = pd.Series((sbl.index > '1983-01-01').astype(int),
+    #                  index=sbl.index).diff(12).iloc[12:]
+
+    # res, _ = gls(endog, exog, order=(0, 0, 12), max_iter=3)
+
+    # assert_allclose(res.exog_params, -328.45, atol=1e-2)
+    # assert_allclose(res.ma_params,
+    #                 [.219, .098, .031, .064, .069, .111, .081,
+    #                  .057, .092, -0.28, .183, -.672], atol=1e-3)
+    # assert_allclose(res.sigma2, 12581, atol=1)
```

Comment:
don't leave commented-out code

Reply:
Good point, I'll remove it and make an issue instead.